notebooks/cookbook/Enhanced sampling methods/Umbrella sampling MD.ipynb

```diff
@@ -0,0 +1,436 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Imports and settings"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Warning: importing 'simtk.openmm' is deprecated.  Import 'openmm' instead.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from openmm.app import *\n",
+    "from openmm import *                    \n",
+    "from openmm.unit import *\n",
+    "from openmmtools import forces\n",
+    "import mdtraj as md\n",
+    "import numpy as np\n",
+    "import parmed as pmd\n",
+    "import bz2\n",
+    "import os\n",
+    "from openmm import CustomIntegrator\n",
+    "from openmm.unit import kilojoules_per_mole, is_quantity\n",
+    "from openmm.unit import *\n",
+    "import numpy as np\n",
+    "import pandas as pd \n",
+    "from matplotlib import pyplot as plt"
```

Comment:
There are a few problems with these imports.

1. Many of them are never used!  So why import them?
2. Cookbook entries need to run in CI.  That means if an external package doesn't get automatically installed with OpenMM, you need to explicitly install it as part of the notebook.
3. For that reason, we should be very sparing about using external packages in the cookbook.  They add complexity and should only be used when there's a really good reason.

Reply:
Makes sense - for clarity, I used the external packages `pandas` and `matplotlib` in other notebooks to demonstrate how one would choose hyperparameters for things like AMD or Metadynamics.

From a teaching standpoint do you think that's strictly necessary or would it be fine to just describe the process and skip ahead? If you think it's strictly necessary I can modify the notebook accordingly to install needed packages.

Reply:
It comes down to what you're trying to teach.  If pandas is central to what you're teaching, then it's necessary.  But if it's just incidental, a way of doing some setup, it's better to omit it.

In the Gaussian AMD tutorial, you use pandas to load a log file from an earlier simulation, then compute a few statistics of the energy from it.  For a cookbook entry, I think it's fine to skip that and just hardcode them.  Something like this:

> Before running a Gaussian AMD simulation, you first need to select a strength for the boost potential.  This is usually done by running a short simulation, recording the potential energy, and computing some simple statistics of it.  The following are the minimum, maximum, mean, and standard deviation of the energies observed during an earlier simulation.

And then just provide hardcode values:

```python
Vmin = -801890.7367725638,
Vmax = -794737.8288112064,
Vavg = -798419.040535957,
Vstd = 904.4602237453533
```